### ISSUE TITLE
feat: add option to save segmentation masks during inference

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/nnBenchmark#16]
## Feature Request

Add a command-line option to save predicted segmentation masks when running `nnBench.inference`.

## Current Behavior

Currently, `nnBench.inference` computes metrics on the test/validation set but does not save the predicted segmentation masks. Users can only see aggregate metrics in `test_history.json`.

## Proposed Behavior

Add a `--save-predictions` flag to save predicted segmentation masks to disk during inference.

```bash
nnBench.inference --config fold_0.yaml --dataset Dataset001_Cellpose --save-predictions
```

## Use Cases

1. **Visual inspection**: Examine individual predictions to understand model behavior
2. **Error analysis**: Identify which samples the model struggles with
3. **Downstream processing**: Use predictions for further analysis or applications
4. **Comparison**: Compare predictions across different models or loss functions
5. **Debugging**: Verify that inference is working correctly

### CLI Option
```bash
nnBench.inference --config <config> --dataset <dataset> [--save-predictions] [--output-dir <path>]
```

Options:
- `--save-predictions`: Enable saving of predicted masks
- `--output-dir <path>`: Optional custom output directory (default: `results_dir/predictions/`)

### Output Structure
```
{results_dir}/
└── predictions/
    ├── case_001.nii.gz
    ├── case_002.nii.gz
    └── ...
```

### Implementation Details
- Save predictions in the same format as input (NIfTI for 3D, PNG/NPY for 2D)
- Preserve original spacing and affine transforms for medical images
- Save as integer class labels (argmax of softmax probabilities)
- Optionally save probability maps with `--save-probabilities` flag

## Additional Options (Future)

- `--save-probabilities`: Save softmax probability maps instead of argmax
- `--save-overlay`: Generate overlay visualizations of predictions on images
- `--save-format`: Specify output format (nifti, npy, png, etc.)

## Related

This complements the existing visualization features but provides full-resolution outputs rather than slice-based visualizations.
